admin-api-server/src/analytics/service/analytics.service.ts

```diff
@@ -258,38 +285,58 @@ LIMIT ${params.pageSize}
-          Number(row['price']),
-          BASE_CURRENCY,
-        );
+      data: await Promise.all(
+        qryRes.rows.map(async (row: any) => {
+          const price = this.currencyService.convertToCurrency(
+            Number(row['price']),
+            BASE_CURRENCY,
+          );
+          const historicRates = await this.currencyService.ratesAt(
+            row['timestamp'],
+          );
+          const feeInBaseCurrency: number =
+            this.currencyService.convertFromCurrency(
+              Number(row['mutez_fee']) / 1000000,
+              'XTZ',
+              historicRates,
+              false,
+            );
+          const nftPriceSumBase = this.currencyService.convertToCurrency(
+            row['nft_price_sum'],
+            BASE_CURRENCY,
+          );
+          const conversionRate =
+            row['nft_order_price'] &&
+            Number(nftPriceSumBase) / row['nft_order_price'];
 
-        const nftPriceSumBase = this.currencyService.convertToCurrency(
-          row['nft_price_sum'],
-          BASE_CURRENCY,
-        );
-        const conversionRate =
-          row['nft_order_price'] &&
-          Number(nftPriceSumBase) / row['nft_order_price'];
-
-        return <Activity>{
-          id: Number(row['id']),
-          timestamp: Math.floor(row['timestamp'].getTime() / 1000),
-          kind: row['kind'],
-          from: row['from'],
-          to: row['to'],
-          tokenId: Number(row['tokenId']),
-          price,
-          edition_size: Number(row['edition_size']),
-          currency: row['currency'],
-          transaction_value:
-            price &&
-            conversionRate &&
-            (Number(price) / conversionRate).toFixed(2),
-          conversion_rate: conversionRate && conversionRate.toFixed(2),
-        };
-      }),
-      count: Number(qryRes.rows[0]['total_activity_count']),
+          return <Activity>{
+            id: Number(row['id']),
+            timestamp: Math.floor(row['timestamp'].getTime() / 1000),
+            kind: row['kind'],
+            from: row['from'],
+            to: row['to'],
+            tokenId: Number(row['tokenId']),
+            price,
+            edition_size: Number(row['edition_size']),
+            currency: row['currency'],
+            transaction_value:
+              price &&
+              conversionRate &&
+              (Number(price) / conversionRate).toFixed(2),
+            conversion_rate: conversionRate && conversionRate.toFixed(2),
+            purchaser_country: row['purchaser_country'],
+            fee_in_base_currency:
+              feeInBaseCurrency > 0
+                ? Number(
+                    this.currencyService
+                      .convertFromBaseUnit(BASE_CURRENCY, feeInBaseCurrency)
+                      .toFixed(4),
+                  )
+                : null,
+          };
+        }),
+      ),
+      count: Number(qryRes.rows[0]['selected_activities_count']),
```

Comment:
I believe this count should be the _total_ count, not the number of returned entries. This is what react-admin then uses to figure out how many additional pages there are.

Reply:
Because of the date range filter, we rarely can navigate between all results. React-admin would show too many pages that don't work if we use the total count here. E.g. the total count could be 300 but we only return 50 because of the date range. Then you only want to see the pages for those 50 and not for all 300, as the date range prevents you from seeing more entries. Let me know if I misunderstand

Reply:
You are right, I am wrong. Nothing to be changed here.